src/middlewared/middlewared/plugins/mdns.py

```diff
@@ -765,10 +766,11 @@ def __init__(self, **kwargs):
 
     def setup(self):
         webui = self.middleware.call_sync('datastore.query', 'system.settings')
-        if (webui[0]['stg_guiprotocol'] == 'http' or
-           webui[0]['stg_guiprotocol'] == 'httphttps'):
+        if not webui[0]['stg_guihttpsredirect']:
```

Comment:
This service should always be advertised on port 6000 (and should have always been). It would be confusing to see advertised port changing after changing irrelevant setting.